lib/registries/addon/drafts/draft/-components/register/component.ts

```diff
@@ -20,6 +21,16 @@ export default class Register extends Component.extend({
             this.setProperties({ rootNode });
         }
     }).on('didReceiveAttrs').restartable(),
+    fetchNodeChildrenRelatedCounts: task(function *(this: Register) {
```

Comment:
This task doesn't just fetch a related count; it's what you call to display the dialog. It'd make more sense to name it `onClickRegister` or similar, move the rest of the `onRegister` action in here, and use `@onClick={{perform this.onClickRegister}}`. No need for a separate action and task.